sdk/src/compute_budget.rs

```diff
@@ -25,8 +24,9 @@ pub enum ComputeBudgetInstruction {
     RequestUnits {
         /// Units to request
         units: u32,
-        /// Additional fee to add
-        additional_fee: u32,
+        /// Additional fee in lamports to charge the payer, used for transaction
+        /// prioritization
+        additional_fee_lamports: u64,
```

Comment:
If we intend to support u32 fees for awhile to avoid breaking clients that are already setting the value, can we keep both versions of `RequestUnits` to make it easier on clients to distinguish the two? Otherwise this will be the only native program instruction which supports multiple encodings and it will be a bit of a headache for web3/explorer and other clients.

Reply:
To be more specific, I would like the enum variant encoded as `1u8` to only support a `u32` fee. Your change would require contents to add special handling for the `1u8` variant to check if it's a `u32` or `u64`. It'd be nicer to add an additional variant (`2u8`) which supports a `u64` fee.

Reply:
If you're onboard with adding a new variant for the u64 fee, I think it would be a good time to split apart "units" and "additional fee". It's a little annoying on the client to have to specifically request compute units when they just want to add a fee and want to use the default compute.

Reply:
We would like to recommend clients to always request units rather than relying on the default values.  Also, splitting these up now requires two instructions to two things most clients should be requesting.

Reply:
Adding a separate instruction to the same program only requires 3 extra bytes to the serialized transaction so I don't think there's any harm in having clients add two separate instructions. The way it's setup now is confusing and bad UX because there's no way to fallback to the default compute units if you want to add an additional fee

Reply:
Ok, yeah, that's fair, I'll update

Reply:
I've been thinking that we're probably going to run into this problem continuously into the future and that a single instruction that accepts an array of compute budget config knobs might be the better option, from the first-three-ix limit perspective

Reply:
Jack and I discussed offline and the first-three-ix issue came up and he proposed just removing that limit and scanning all instructions. I also proposed that we have the feature gate in this PR do 2 things:
1. Update `RequestUnits` to just one u32 for compute units (remove fee here)
2. Add new `AdditionalFee` instruction which supports a `u64` for the lamport fee

Reply:
mmm, no concerns over removing the first-three-ix logic in light of a single MTU tx being able to fit 350+ instructions?  sounds like a lot of pubkey comparisons

Reply:
Meh, I think pubkey comparisons are the least of our problems in that case :)

Reply:
probably not?  pretty sure you could NOP them into the minimum CU fee bucket. worse when we extend max tx size

Reply:
Transaction account keys are all unique, we only need to find the compute program id once and see where it's used. I really don't think this is going to be an issue

Reply:
> Jack and I discussed offline and the first-three-ix issue came up and he proposed just removing that limit and scanning all instructions. I also proposed that we have the feature gate in this PR do 2 things (in addition to scrapping the first-three-ix logic):
> 
> 1. Update `RequestUnits` to just one u32 for compute units (remove fee here)
> 2. Add new `AdditionalFee` instruction which supports a `u64` for the lamport fee

I agree with this approach, one thing is that it will break SDK backward compatibility, though the only ones using it for now is probably neon on testnet/devnet for testing, I'll alert them